src/scanner/present.rs

```diff
@@ -171,44 +175,134 @@ async fn handle_active_oracles_answering(
             Ok(oracles) => oracles,
             Err(error) => {
                 tracing::error!(
-                    "could not get currently active oracles in chain with id {} - {}",
+                    "could not get currently active oracles in chain with id {} - {:#}",
                     context.chain_id,
                     error
                 );
-                return;
+                return Ok(());
             }
         };
 
+    let active_oracles_len = active_oracles.len();
+    if active_oracles_len > 0 {
+        tracing::info!("trying to answer {} active oracles", active_oracles_len);
+    }
+
     let mut join_set: JoinSet<Result<(), anyhow::Error>> = JoinSet::new();
     for active_oracle in active_oracles.into_iter() {
         let chain_id = context.chain_id;
         join_set.spawn(
             answer_active_oracle(
                 signer.clone(),
                 context.db_connection_pool.clone(),
+                context.defillama_client.clone(),
                 active_oracle,
             )
             .instrument(error_span!("answer", chain_id)),
         );
     }
+
+    // wait forever unless some task stops with an error
+    while let Some(res) = join_set.join_next().await {
+        let _ = res.context("task unexpectedly stopped")?;
```

Comment:
Is the `let _ =` here needed?

Reply:
Yessir, otherwise we have a warning regarding the `Result` not being used